src/contracts/middleware/BLSRegistryCoordinatorWithIndices.sol

```diff
@@ -0,0 +1,325 @@
+        registries.push(address(blsPubkeyRegistry));
+        registries.push(address(indexRegistry));
+
+        // set the operator set params
+        require(IVoteWeigher(address(stakeRegistry)).quorumCount() == _operatorSetParams.length, "BLSIndexRegistryCoordinator: operator set params length mismatch");
+        for (uint8 i = 0; i < _operatorSetParams.length; i++) {
+            quorumOperatorSetParams[i] = _operatorSetParams[i];
+        }
+    }
+
+    /// @notice Returns task number from when `operator` has been registered.
+    function getFromTaskNumberForOperator(address operator) external view returns (uint32) {
+        return _operators[operator].fromTaskNumber;
+    }
+
+    /// @notice Returns the operator struct for the given `operator`
+    function getOperator(address operator) external view returns (Operator memory) {
+        return _operators[operator];
+    }
+
+    /// @notice Returns the operatorId for the given `operator`
+    function getOperatorId(address operator) external view returns (bytes32) {
+        return _operators[operator].operatorId;
+    }
+
+    /// @notice Returns the quorum bitmap for the given `operatorId` at the given `blockNumber` via the `index`
+    function getQuorumBitmapByOperatorIdAtBlockNumberByIndex(bytes32 operatorId, uint32 blockNumber, uint256 index) external view returns (uint192) {
+        QuorumBitmapUpdate memory quorumBitmapUpdate = _operatorIdToQuorumBitmapHistory[operatorId][index];
+        require(
+            quorumBitmapUpdate.updateBlockNumber <= blockNumber, 
+            "BLSRegistryCoordinator.getQuorumBitmapByOperatorIdAtBlockNumberByIndex: quorumBitmapUpdate is from after blockNumber"
+        );
+        require(
+            quorumBitmapUpdate.nextUpdateBlockNumber > blockNumber, 
+            "BLSRegistryCoordinator.getQuorumBitmapByOperatorIdAtBlockNumberByIndex: quorumBitmapUpdate is from before blockNumber"
+        );
+        return quorumBitmapUpdate.quorumBitmap;
+    }
+
+    /// @notice Returns the current quorum bitmap for the given `operatorId`
+    function getCurrentQuorumBitmapByOperatorId(bytes32 operatorId) external view returns (uint192) {
+        return _operatorIdToQuorumBitmapHistory[operatorId][_operatorIdToQuorumBitmapHistory[operatorId].length - 1].quorumBitmap;
+    }
+
+    /// @notice Returns the number of registries
+    function numRegistries() external view returns (uint256) {
+        return registries.length;
+    }
+
+    /**
+     * @notice Sets parameters of the operator set for the given `quorumNumber`
+     * @param quorumNumber is the quorum number to set the maximum number of operators for
+     * @param operatorSetParam is the parameters of the operator set for the `quorumNumber`
+     */
+    function setOperatorSetParams(uint8 quorumNumber, OperatorSetParam memory operatorSetParam) external onlyServiceManagerOwner {
+        quorumOperatorSetParams[quorumNumber] = operatorSetParam;
```

Comment:
OK yeah so another place for the internal setter function w/ checks + event here.